iceoryx_dds/CMakeLists.txt

```diff
@@ -132,41 +100,24 @@ if(USE_CYCLONE_DDS)
     )
 endif()
 
-target_compile_options(iceoryx_dds PRIVATE ${ICEORYX_WARNINGS} ${ICEORYX_SANITIZER_FLAGS})
-
 #
 ########## build gateway app ##########
 #
-add_executable(iox-dds-gateway
-    source/gateway/main.cpp
-)
-target_link_libraries(iox-dds-gateway
-    iceoryx_posh::iceoryx_posh
-    iceoryx_posh::iceoryx_posh_gateway
-    iceoryx_posh::iceoryx_posh_config
-    ${PROJECT_NAMESPACE}::iceoryx_dds
-)
-
-target_compile_options(iox-dds-gateway PRIVATE ${ICEORYX_WARNINGS} ${ICEORYX_SANITIZER_FLAGS})
-
-set_target_properties(iox-dds-gateway iceoryx_dds
-  PROPERTIES
-    CXX_STANDARD_REQUIRED ON
-    CXX_STANDARD ${ICEORYX_CXX_STANDARD}
-)
+iox_add_executable(
+    NO_PACKAGE_SETUP
+    NO_FIND_PACKAGE_SUPPORT
+    TARGET              iox-dds-gateway
+    LIBS                iceoryx_posh::iceoryx_posh
+                        iceoryx_posh::iceoryx_posh_gateway
+                        iceoryx_posh::iceoryx_posh_config
+                        iceoryx_dds::iceoryx_dds
```

Comment:
I suppose not everything is needed here?

Reply:
We use those libraries directly therefore we require them here.